src/main/java/ChessBoardUI.java

```diff
@@ -0,0 +1,101 @@
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.util.AbstractMap;
+import java.util.Map;
+
+public class ChessBoardUI extends JPanel {
+    private LocationBitboard locationBitboard;
+    public int SQUARE_SIZE = 85;
+    private Map.Entry<String, Long> draggedPiece = null;
+
+    public ChessBoardUI() {
+        this.locationBitboard = new LocationBitboard();
+
+        addMouseListener(new MouseAdapter() {
+            @Override
+            public void mousePressed(MouseEvent e) {
+                int x = e.getX() / SQUARE_SIZE;
+                int y = 7 - e.getY() / SQUARE_SIZE;
+                int index = y * 8 + x;
+
+                // Iterate over all piece types to find which piece attribute was clicked
+                for (String pieceType : locationBitboard.getAllPieces().keySet()) {
+                    long bitboard = locationBitboard.getBitboard(pieceType);
+                    if ((bitboard & (1L << index)) != 0) {
+                        bitboard &= ~(1L << index);  // Clear the bit at the selected index
+                        locationBitboard.setBitboard(pieceType, bitboard);
+                        draggedPiece = new AbstractMap.SimpleEntry<>(pieceType, bitboard);
```

Comment:
I see we are making the piece we click "disappear" while it is clicked and using draggedPiece to make it appear again in mouseReleased method below.